### PR TITLE
[ExternalBehaviorModel] Without compatibility layer

### PR DESCRIPTION
**Pull request by alxbilger**
Also, I made `ExternalBehaviorModel` depends on `SOFA_ENABLE_LEGACY_HEADERS`, since it depends on `SofaGraphComponent` which also depends on `SOFA_ENABLE_LEGACY_HEADERS`

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: applications/plugins/ExternalBehaviorModel/CMakeLists.txt [hunk 1]
```diff
@@ -14,16 +14,15 @@ set(SOURCE_FILES
...
+sofa_find_package(Sofa.Component.Constraint.Projective REQUIRED)
 sofa_find_package(SofaGraphComponent REQUIRED)
-sofa_find_package(SofaImplicitOdeSolver REQUIRED)
-sofa_find_package(SofaSimpleFem REQUIRED)
+sofa_find_package(Sofa.Component.ODESolver.Backward REQUIRED)
+sofa_find_package(Sofa.Component.SolidMechanics.FEM.Elastic REQUIRED)
 
 add_library(${PROJECT_NAME} SHARED ${HEADER_FILES} ${SOURCE_FILES})
-target_link_libraries(${PROJECT_NAME} SofaBaseLinearSolver SofaBaseMechanics SofaBaseTopology SofaBoundaryCondition SofaGraphComponent SofaImplicitOdeSolver SofaSimpleFem)
+target_link_libraries(${PROJECT_NAME} Sofa.Component.LinearSolver.Iterative SofaBaseMechanics Sofa.Component.Topology Sofa.Component.Constraint.Projective SofaGraphComponent Sofa.Component.ODESolver.Backward Sofa.Component.SolidMechanics.FEM.Elastic)
```
Comment (fredroy) [inline applications/plugins/ExternalBehaviorModel/CMakeLists.txt]:
```suggestion
target_link_libraries(${PROJECT_NAME} Sofa.Component.LinearSolver.Iterative Sofa.Component.Topology Sofa.Component.Constraint.Projective SofaGraphComponent Sofa.Component.ODESolver.Backward Sofa.Component.SolidMechanics.FEM.Elastic)
```

**Review by fredroy — APPROVED** [review]

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

**Comment by damienmarchal** [comment]
Ready, merge if CI validates it. An issue should be open to point the dependency on the deprecated SofaGraphComponent.

**Comment by alxbilger** [comment]
The issue is https://github.com/sofa-framework/sofa/issues/3319

**Pull request merged**